lib/find.ts

```diff
@@ -1,32 +1,42 @@
 import * as fs from 'fs';
 import { join } from 'path';
-import { promisify } from 'util';
 import { debug } from './debug';
 
-const readdir = promisify(fs.readdir);
-const stat = promisify(fs.stat);
+const MAX_FILE_SIZE: number = 2 * 1024 * 1024 * 1024;
 
-export async function find(dir: string): Promise<string[]> {
-  const result: string[] = [];
+export interface FsHandler {
+  stat(path: fs.PathLike): fs.Stats;
+  readdir(path: string): string[];
+}
+
+const defaultFsHandler: FsHandler = {
+  stat: fs.statSync,
+  readdir: fs.readdirSync,
+};
 
-  const dirStat = await stat(dir);
+export function find(
+  dir: string,
+  fsHandler: FsHandler = defaultFsHandler,
```

Comment:
I think adding another level of interface to deal with multiple fsHandlers is too much as we are dealing only with a single readfile "handler" case.

Reply:
I wanted to make it testable, without introducing mock libraries that make use of magic behind the scene. That's the reason

Reply:
I understand your point, albeit we already have an installed jest library with built-in mock capabilities that allow us to the job and is completely fine to use for testing. We would not be introducing anything different except for the spyOn on test file

Reply:
(or to rephrase it, usually I prefer polymorphism (via parameters) if the alternative is mock-libs. That doesn't mean I don't make use of mocks, but it's usually not my preferred solution)